### PR TITLE
Show factory queue on rollover

### PR DESCRIPTION
**Pull request by 4z0t**
Shows factory queue on top of unit info on rollover
fac queue4

Edit by Jip:
 - [x] Add game option to disable

### CODE REVIEW

**Comment by Garanas** [comment]
Initial tests appear to work fine: as a player I can not see the queue of anyone else (allied or enemy), I can see my own. As an observer I can see all the queues by hovering.

**Comment by Garanas** [comment]
I'll test again tomorrow and review the code.

**Comment by 4z0t** [comment]
@Garanas also, mb need to change the way we check if unit is a factory, it looks ugly rn 😐

**Comment by Garanas** [comment]
I'll switch that up 😄

File: lua/ui/game/unitview.lua [hunk 1]
```diff
@@ -476,6 +520,14 @@ function UpdateWindow(info)
         if info.userUnit then
             unitQueue = info.userUnit:GetCommandQueue()
         end
+        CreateQueueGrid(controls.bg)
+        --TODO: unit category better check!!!!!!!!!
+        if not table.empty(EntityCategoryFilterDown((categories.SHOWQUEUE * categories.STRUCTURE) + categories.FACTORY,
+        {info.userUnit})) and
+        selectedUnit~= info.userUnit then
```
Comment (Garanas) [inline lua/ui/game/unitview.lua:527]: Why this particular check? I think for consistency reasons it doesn't hurt

**Comment by Garanas** [comment]
It also works fine when dragging the queue / changing the queue in some fashion

File: lua/ui/game/unitview.lua [hunk 2]
```diff
@@ -476,6 +522,15 @@ function UpdateWindow(info)
         if info.userUnit then
             unitQueue = info.userUnit:GetCommandQueue()
         end
+        CreateQueueGrid(controls.bg)
```
Comment (Garanas) [inline lua/ui/game/unitview.lua:525]: Shouldn't this be inside the if statement too?

**Comment by Marc-Spector** [comment]
It would be nice to add the ability to disable this feature in the settings. Not everyone has large screens.

**Comment by Garanas** [comment]
Good suggestion - I'll add that in.

File: loc/RU/strings_db.lua [hunk 3]
```diff
@@ -8595,6 +8595,7 @@ OPTIONS_0242="Всегда показывать имена"
 OPTIONS_0243="Показывать вражеские индикаторы жизни"
 OPTIONS_0244="Показывать вооружение в строит-ом меню"
 OPTIONS_0245="Улучшенный сброс выбранных юнитов"
+OPTIONS_0246 = "Show Factory Queue on Hover"
```
Comment (Garanas) [inline loc/RU/strings_db.lua:8598]: @4z0t can you make this a proper russian translation?

**Comment by Garanas** [comment]
@Marc-Spector done!

**Comment by speed2CZ** [comment]
While the idea of showing the preview of the queue is good, these wild icons slapped above the window look bad.

**Comment by AigizIskuzhin** [comment]
Grid ui can be visible on first hover
- game launched
- no factory under mouse
image
- on first hover on factory it being created and you can notice it
image
- now it is gone
image

**Comment by AigizIskuzhin** [comment]
Why queue of factory passed on to the ACU when pointing acu?

test

**Comment by Garanas** [comment]
Perhaps we should hide it when we have the factory selected then 😄 .

**Comment by Garanas** [comment]
image

Final state of the UI

**Comment by Garanas** [comment]
Tested with scaling the UI, turning on / off various settings. All appears to be good.

**Comment by 4z0t** [comment]
>> add always show in obs option

**Comment by Garanas** [comment]
I'll add that to #3604 .

**Pull request merged**